### PR TITLE
Scaffold generators

### PR DESCRIPTION
**Pull request by Hinton**
```
- [ ] Bug fix
- [x] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Scaffold the password and passphrase generator. Wire up a command for password in the bw cli.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)

### CODE REVIEW

File: crates/bw/src/main.rs [hunk 1]
```diff
@@ -33,6 +33,12 @@ enum Commands {
 
     #[command(long_about = "Pull the latest vault data from the server")]
     Sync {},
+
+    #[command(long_about = "Pull the latest vault data from the server")]
```
Comment (dani-garcia) [inline crates/bw/src/main.rs:37]: This comment needs to be updated

File: crates/bw/src/main.rs [hunk 2]
```diff
@@ -62,6 +68,39 @@ enum ItemCommands {
...
+        long,
+        action,
+        help = "Include special characters (!@#$%^&*)"
+    )]
+    special: bool,
+
+    #[arg(
+        long,
+        default_value = "16",
+        help = "Minimum length of generated password"
```
Comment (dani-garcia) [inline crates/bw/src/main.rs:99]: There's no minimum and maximum, the length of the generated password is exactly what's provided, so the comment should probably be "Length of generated password" right?

File: crates/bw/src/main.rs [hunk 3]
```diff
@@ -62,6 +68,39 @@ enum ItemCommands {
...
+#[derive(Args, Clone)]
+struct PasswordGeneratorArgs {
+    #[arg(short = 'l', long, action, help = "Include lowercase characters (a-z)")]
+    lowercase: bool,
+
+    #[arg(short = 'u', long, action, help = "Include uppercase characters (A-Z)")]
+    uppercase: bool,
+
+    #[arg(short = 'n', long, action, help = "Include numbers (0-9)")]
+    numbers: bool,
```
Comment (dani-garcia) [inline crates/bw/src/main.rs:86]: Not sure about these parameters, by default they are all set to false, so running `bw generate password` we won't have any character types selected, which would probably have to return an error. 

You'll have to go out of your way to do `bw generate password -l -u -n` to get an alphanumeric password.

I think having a reasonable default for `bw generate password` is a good idea. Maybe we should invert the flags and make them `no-lowercase`, `no-uppercase` and `no-numbers`?

File: crates/bw/src/main.rs (hunk 2 above)
Comment (Hinton) [reply crates/bw/src/main.rs:99]: Yep, codepilot 😠

File: crates/bw/src/main.rs (hunk 3 above)
Comment (Hinton) [reply crates/bw/src/main.rs:86]: My thinking is that if all are false the default applies. Which is how the CLI currently works.

**Review by dani-garcia — APPROVED** [review]

**Pull request merged**